slack.js

```diff
@@ -0,0 +1,266 @@
+//------------------------------------------------------------------------------
+// Bot / Slack related helpers
+//------------------------------------------------------------------------------
+var Botkit = require('botkit');
```

Comment:
the more and more I read this code, the more I think internalizing the botkit api would be a good idea.

it will simplify the interface if the client doesnt need to know about botkit and we can wrap up all the botkit specific code within one file.

What are your thoughts?